src/actions/drag_mover.ts

```diff
@@ -0,0 +1,201 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Constants from '../constants';
+import {
+  ASTNode,
+  Connection,
+  ShortcutRegistry,
+  WorkspaceSvg,
+  common,
+  registry,
+  utils,
+} from 'blockly';
+import type {Block, BlockSvg, IDragger} from 'blockly';
+import {Mover, MoveInfo} from './mover';
+import {Navigation} from '../navigation';
+
+/**
+ * An experimental implementation of Mover that uses a dragger to
+ * perform unconstraind movment.
+ */
+export class DragMover extends Mover {
+  /**
+   * The distance to move an item, in workspace coordinates, when
+   * making an unconstrained move.
+   */
+  UNCONSTRAINED_MOVE_DISTANCE = 20;
+
+  /**
+   * Map of moves in progress.
+   *
+   * An entry for a given workspace in this map means that the this
+   * Mover is moving a block on that workspace, and will disable
+   * normal cursor movement until the move is complete.
+   */
+  protected declare moves: Map<WorkspaceSvg, DragMoveInfo>;
+
+  /**
+   * Start moving the currently-focused item on workspace, if
+   * possible.
+   *
+   * Should only be called if canMove has returned true.
+   *
+   * @param workspace The workspace we might be moving on.
+   * @returns True iff a move has successfully begun.
+   */
+  override startMove(workspace: WorkspaceSvg) {
+    const cursor = workspace?.getCursor();
+    const curNode = cursor?.getCurNode();
+    const block = curNode?.getSourceBlock() as BlockSvg | null;
+    if (!cursor || !block) throw new Error('precondition failure');
+
+    // Select and focus block.
+    common.setSelected(block);
+    cursor.setCurNode(ASTNode.createBlockNode(block)!);
+    // Begin dragging block.
+    const DraggerClass = registry.getClassFromOptions(
+      registry.Type.BLOCK_DRAGGER,
+      workspace.options,
+      true,
+    );
+    if (!DraggerClass) throw new Error('no Dragger registered');
+    const dragger = new DraggerClass(block, workspace);
+    // Record that a move is in progress and start dragging.
+    const info = new DragMoveInfo(block, dragger);
+    this.moves.set(workspace, info);
+    // Begin drag.
+    dragger.onDragStart(info.fakePointerEvent('pointerdown'));
+    return true;
+  }
+
+  /**
+   * Finish moving the currently-focused item on workspace.
+   *
+   * Should only be called if isMoving has returned true.
+   *
+   * @param workspace The workspace on which we are moving.
+   * @returns True iff move successfully finished.
+   */
+  override finishMove(workspace: WorkspaceSvg) {
+    if (!workspace) return false;
```

Comment:
As in the base class, either remove the null check or make the arg nullable here and in `abortMove`